samples/bluetooth/central_otc/src/main.c

```diff
@@ -0,0 +1,585 @@
+			 struct net_buf_simple *ad)
+{
+	char dev[BT_ADDR_LE_STR_LEN];
+
+	bt_addr_le_to_str(addr, dev, sizeof(dev));
+
+	/* We're only interested in connectable events and scan response
+	 * because service UUID is in sd of sample peripheral_ots.
+	 */
+	if (type == BT_GAP_ADV_TYPE_ADV_IND || type == BT_GAP_ADV_TYPE_ADV_DIRECT_IND ||
+	    type == BT_GAP_ADV_TYPE_SCAN_RSP) {
+		bt_data_parse(ad, eir_found, (void *)addr);
+	}
+}
+
+static void start_scan(void)
+{
+	int err;
+
+	/* Use active scanning and disable duplicate filtering to handle any
+	 * devices that might update their advertising data at runtime.
+	 */
+	struct bt_le_scan_param scan_param = {
+		.type = BT_LE_SCAN_TYPE_ACTIVE,
+		.options = BT_LE_SCAN_OPT_NONE,
+		.interval = BT_GAP_SCAN_FAST_INTERVAL,
+		.window = BT_GAP_SCAN_FAST_WINDOW,
+	};
+
+	err = bt_le_scan_start(&scan_param, device_found);
+	if (err) {
+		printk("Scanning OTS TAG failed to start (err %d)\n", err);
+		return;
+	}
+
+	printk("Scanning successfully started\n");
+}
+
+static void subscribe_func(void)
+{
+	printk("subscribe OACP and OLCP Indication\n");
+	oacp_sub_params = &otc.oacp_sub_params;
+	oacp_sub_params->disc_params = &otc.oacp_sub_disc_params;
+	if (oacp_sub_params) {
+		/* With ccc_handle == 0 it will use auto discovery */
+		oacp_sub_params->ccc_handle = 0;
+		oacp_sub_params->end_handle = otc.end_handle;
+		oacp_sub_params->value = BT_GATT_CCC_INDICATE;
+		oacp_sub_params->value_handle = otc.oacp_handle;
+		oacp_sub_params->notify = bt_ots_client_indicate_handler;
+		bt_gatt_subscribe(default_conn, oacp_sub_params);
+	}
+	olcp_sub_params = &otc.olcp_sub_params;
+	olcp_sub_params->disc_params = &otc.olcp_sub_disc_params;
+	if (olcp_sub_params) {
+		/* With ccc_handle == 0 it will use auto discovery */
+		olcp_sub_params->ccc_handle = 0;
+		olcp_sub_params->end_handle = otc.end_handle;
+		olcp_sub_params->value = BT_GATT_CCC_INDICATE;
+		olcp_sub_params->value_handle = otc.olcp_handle;
+		olcp_sub_params->notify = bt_ots_client_indicate_handler;
+		bt_gatt_subscribe(default_conn, olcp_sub_params);
+	}
+}
+
+static bool is_discovery_complete(void)
+{
+	return (atomic_test_bit(&discovery_state, DISC_OTS_FEATURE) &&
+		atomic_test_bit(&discovery_state, DISC_OTS_NAME) &&
+		atomic_test_bit(&discovery_state, DISC_OTS_TYPE) &&
+		atomic_test_bit(&discovery_state, DISC_OTS_SIZE) &&
+		atomic_test_bit(&discovery_state, DISC_OTS_ID) &&
+		atomic_test_bit(&discovery_state, DISC_OTS_PROPERTIES) &&
+		atomic_test_bit(&discovery_state, DISC_OTS_ACTION_CP) &&
+		atomic_test_bit(&discovery_state, DISC_OTS_LIST_CP));
+}
+
+static uint8_t discover_func(struct bt_conn *conn, const struct bt_gatt_attr *attr,
+			     struct bt_gatt_discover_params *params)
+{
+	int err;
+
+	if (!attr) {
+		printk("Discover complete\n");
+		(void)memset(params, 0, sizeof(*params));
+		return BT_GATT_ITER_STOP;
+	}
+
+	if (!bt_uuid_cmp(discover_params.uuid, BT_UUID_OTS)) {
+		memcpy(&uuid, BT_UUID_OTS_FEATURE, sizeof(uuid));
```

Comment:
Generally in Zephyr we `(void)` cast the return values of `memcpy` and `memset`
```suggestion
		(void)memcpy(&uuid, BT_UUID_OTS_FEATURE, sizeof(uuid));
```